### PR TITLE
Add VLC audio backend

### LINKED ISSUE #18: Support M4A format
Support playback of .m4a file format

### LINKED ISSUE #19: Support AAC
Support playback of .aac file format

### LINKED ISSUE #20: Suport FLAC
Support playback of .flac file format

### PR DESCRIPTION
**Pull request by mzlarsson**
Seems like the pygame lib might have limitations. Added an alternative audio backend

### CODE REVIEW

**Comment by mzlarsson** [comment]
Note about diff: I renamed `audio.py` to `audio_pygame.py` (only added line 51) but git didn't realize. So those are practically identical.

File: audio/src/audio_vlc.py [hunk 1]
```diff
@@ -0,0 +1,139 @@
+import os
+import time
+import threading
+os.add_dll_directory(r"C:\Program Files (x86)\VideoLAN\VLC")
```
Comment (AntonNeld) [inline audio/src/audio_vlc.py]: This looks platform-specific, what will happen on Linux?

**Review by AntonNeld — COMMENTED** [review]
I think you also need to add python-vlc to the Pipfile.

What are the limitations of pygame, and is there still a reason to keep it as an alternative?

File: audio/src/audio_vlc.py (hunk 1 above)
Comment (mzlarsson) [reply audio/src/audio_vlc.py]: Oh true, I had that as a ToDo but forgot to fix. My plan is to add multiple paths so they cover all possible OSes we have, does that make sense?

**Comment by mzlarsson** [comment]
> What are the limitations of pygame, and is there still a reason to keep it as an alternative?

Mainly it's that we can't seek in the files (aka jump to a specific timestamp), but the API is rather small so I think we would have found more things. VLC is able to play video files too, so the audio implementation of video clips will be quite easy with VLC.
We could keep pygame as an alternative (but disable some new features, such as seek) but we should probably focus on VLC.

File: audio/src/audio_vlc.py (hunk 1 above)
Comment (AntonNeld) [reply audio/src/audio_vlc.py]: I'm not sure how it would look in practice, but as long as it's handled I think that's fine.

**Comment by AntonNeld** [comment]
If VLC is strictly better, why not remove pygame already in this PR? It would make things simpler to not have the infrastructure for multiple backends.

**Comment by mzlarsson** [comment]
Tested on Windows and Linux (Ubuntu). Works as long as you have VLC installed. I'm not sure the path to plugins is required (didn't exist on my Linux-PC but it still worked) but leaving it there just in case.

**Review by AntonNeld — APPROVED** [review]

**Pull request merged**